src/components/filters/useGroupFilter.js

```diff
@@ -32,39 +30,37 @@ export const buildHostGroupChips = (selectedGroups = []) => {
     : [];
 };
 
-const useGroupFilter = (apiParams = []) => {
-  const dispatch = useDispatch();
+const useGroupFilter = () => {
   const groupsEnabled = useFeatureFlag('hbi.ui.inventory-groups');
+  const { fetchBatched } = useFetchBatched();
+  const [groups, setGroups] = useState([]);
+  const [selected, setSelected] = useState([]);
+
+  const onHostGroupsChange = useCallback((event, selection, item) => {
+    setSelected(union(selection, item));
+  }, []);
 
   useEffect(() => {
-    if (groupsEnabled === true) {
-      dispatch(fetchGroupsForEntities(apiParams)); // TODO: make the request paginated (to show all the groups)
+    const fetchOptions = async () => {
```

Comment:
It's a bit problematic because of usage of `useFetchBatched` hook. Though, I will surely extract it once there are more applications of this function in the future